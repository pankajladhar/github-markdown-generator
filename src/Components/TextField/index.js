import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './TextField.scss';

const propTypes = {
    type: PropTypes.oneOf(['text', 'password', 'search']).isRequired,
    handleChange: PropTypes.func.isRequired,
}
const defaultProps = {    
    className: '',
}

export default class TextField extends PureComponent {
    render() {
        return (
                <input type={this.props.type} 
                    className={this.props.className.trim()}
                    name={this.props.name}
                    placeholder={this.props.placeholder}
                    onChange={this.props.handleChange}
                />
        );
    }
}

TextField.propTypes = propTypes;
TextField.defaultProps = defaultProps;
