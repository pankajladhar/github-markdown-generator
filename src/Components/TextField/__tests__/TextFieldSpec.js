import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TextField from './../index';

describe('TextField', ()=>{
    let textFieldProps ={}

    beforeEach(()=>{
        textFieldProps = {
            type: "text",
            handleChange: ()=>{}
        }
    });

    it('renders correctly when only required props are provided', ()=>{
        let tree = renderer.create(
            <TextField  {...textFieldProps}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when all props are provided', ()=>{
        textFieldProps.name = "textfield";
        textFieldProps.placeholder = "This is placeholder text";
        textFieldProps.className = "sample";

        let tree = renderer.create(
            <TextField  {...textFieldProps}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should call the provided change handler', ()=>{
        
        let mockHandleChange = jest.fn();
        
        textFieldProps.handleChange = mockHandleChange;
        
        const textField = shallow(
            <TextField {...textFieldProps}/>
        );

        textField.simulate('change');
        expect(mockHandleChange).toHaveBeenCalled();
    })

});
