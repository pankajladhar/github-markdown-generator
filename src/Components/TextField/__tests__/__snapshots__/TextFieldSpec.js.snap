// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders correctly when all props are provided 1`] = `
<input
  className="sample"
  name="textfield"
  onChange={[Function]}
  placeholder="This is placeholder text"
  type="text"
/>
`;

exports[`TextField renders correctly when only required props are provided 1`] = `
<input
  className=""
  name={undefined}
  onChange={[Function]}
  placeholder={undefined}
  type="text"
/>
`;
